sway/input/seatop_default.c

```diff
@@ -471,6 +473,43 @@ static void handle_motion(struct sway_seat *seat, uint32_t time_msec,
 	e->previous_node = node;
 }
 
+static void handle_tablet_tool_motion(struct sway_seat *seat,
+		struct sway_tablet *tablet, struct wlr_tablet_tool *tool,
+		uint32_t time_msec, double dx, double dy) {
+	struct seatop_default_event *e = seat->seatop_data;
+	struct sway_cursor *cursor = seat->cursor;
+
+	struct wlr_surface *surface = NULL;
+	double sx, sy;
+	struct sway_node *node = node_at_coords(seat,
+			cursor->cursor->x, cursor->cursor->y, &surface, &sx, &sy);
+
+	if (node && config->focus_follows_mouse != FOLLOWS_NO) {
+		check_focus_follows_mouse(seat, e, node);
+	}
+
+	if (surface) {
+		if (seat_is_input_allowed(seat, surface)) {
+			struct sway_tablet_tool *sway_tool = tool->data;
+			wlr_tablet_v2_tablet_tool_notify_proximity_in(sway_tool->tablet_v2_tool,
+				tablet->tablet_v2, surface);
+			wlr_tablet_v2_tablet_tool_notify_motion(sway_tool->tablet_v2_tool, sx,
+				sy);
+		}
+	} else {
+		cursor_update_image(cursor, node);
```

Comment:
Maybe we should do an equivalent of `wlr_seat_pointer_clear_focus` but for the tablet here?

Reply:
I think the closest equivalent would be `wlr_tablet_v2_tablet_tool_notify_proximity_out`. I did consider calling it here, but in practice it doesn't change the behavior (since it's implicitly called during the next frame anyway). That said, it's idempotent, so might as well be explicit and call it here too.